aiida/cmdline/commands/cmd_storage.py

```diff
@@ -93,13 +93,16 @@ def storage_integrity():
 def storage_info(statistics):
     """Summarise the contents of the storage."""
     from aiida.cmdline.utils.common import get_database_summary
+    from aiida.manage.manager import get_manager
     from aiida.orm import QueryBuilder
-    from aiida.storage.control import get_repository_info
+
+    manager = get_manager()
+    storage = manager.get_profile_storage()
 
     with spinner():
         data = {
             'database': get_database_summary(QueryBuilder, statistics),
-            'repository': get_repository_info(statistics=statistics),
+            'repository': storage.get_info(statistics=statistics),
```

Comment:
This doesn't quite make sense 😬 
`get_info` doctring: "Return general information on the storage.", i.e. not just about the repository

Reply:
I realized this, but considered this out of the scope of this PR, which is just introducing the use of the locking mechanism. I would leave this for another PR, which then also moves `get_database_summary` into the `StorageBackend`.

Reply:
can we open an issue for this then

Reply:
https://github.com/aiidateam/aiida-core/issues/5384

Reply:
Personally, I wouldn't really say it is out of scope, because it is this PR that is introducing the "semantic" bug that was not there before.
But won't block this PR 👍

Reply:
I see @sphuber [already sumarized](https://github.com/aiidateam/aiida-core/issues/5384#issue-1147934582) better than I can the problem with `get_database_summary` using more high level methods to get its information. I'll just mention that I see usefulness both in being able to get some statistics in a "backend" agnostic way, as well as being able to get more details specific to the backend. We might need to just think a bit if there is some structural design that allows to have both.

In the meantime, I generalized `storage.get_info` so now the docstring is correct, it returns the information of the storage: that is, both the `repository` and the `database` (it just happens so that the `database` is currently empty). Then the `storage_info` command just calls `get_database_summary` to add a `summary` key to the `database` after it is returned. I believe this is the way in which we keep all the prior information with minimal changes that still leaves the methods in the best baseline to later re-organize the content of `storage.get_info` and `get_database_summary`.